packages/minter-contracts/ligo/fa2_modules/fa2_allowlist/allowlist_token.mligo

```diff
@@ -0,0 +1,63 @@
+(* Allowlist of FA2 addresses and token_ids.
+ *
+ * This allows permitting only specific addresses, and for some addresses
+ * also restricting the set of allowed token_ids.
+ * This suits for large addresses sets, however token_ids set per each address
+ * is assumed to be small.
+*)
+
+#if !ALLOWLIST_TOKEN
+#define ALLOWLIST_TOKEN
+#define ALLOWLIST_ENABLED
+
+#include "../../fa2/fa2_interface.mligo"
+
+type allowed_token_ids =
+  | All_token_ids_allowed
+  | Token_ids_allowed of token_id set
+
+type allowlist = (address, allowed_token_ids) big_map
+
+type allowlist_update =
+  [@layout:comb]
+  { to_remove : address set
+  ; to_add : (address, allowed_token_ids) map
+  }
+
+type allowlist_entrypoints = allowlist_update
+  // TODO try list/set
```

Comment:
I see you went with a `set`, is this TODO here intended to check performance differences with a `list`?

Reply:
Yeah right, thanks for the spot. Eventually, I tried to replace `(address, allowed_token_ids) map ` with a list of pairs, and that increased fees (:open_mouth:), so I'll avoid `list`s at all.